### PR TITLE
Test scripts

### PR DESCRIPTION
**Pull request by Oddj0b**
I've created examples of how to add App Center Test to Xamarin Android || iOS projects. I'm rusty in bash so I might have missed something.

### CODE REVIEW

File: sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
```
Comment (maxxkrakoa) [inline sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh]: I would add `set -e` here at the top to make sure that the script bails out if there are any statements that return an error.

File: sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+echo
+echo "Building UI test projects:"
+find $APPCENTER_SOURCE_DIRECTORY -name '*.UITest.csproj' -exec msbuild {} \;
+echo "Compiled projects to run UI tests:"
+find $APPCENTER_SOURCE_DIRECTORY -regex '*.bin.*UITest.*\.dll' -exec echo {} \;
+#echo "Running UI test in App Center Test:"
+echo "What is in the output directory"
+ls $APPCENTER_OUTPUT_DIRECTORY
+echo "Running test in App Center Test"
+APPPATH=$APPCENTER_OUTPUT_DIRECTORY/*.apk
```
Comment (maxxkrakoa) [inline sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh]: Add a check here to ensure that the various paths and directories have been correctly found - and bail out with an error if they haven't been found.

File: sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+BUILDDIR=$APPCENTER_SOURCE_DIRECTORY/*.UITest/bin/Debug/
+appcenter test run uitest --app $APP_OWNER --devices $DEVICE_SET --test-series "$APPCENTER_BRANCH-$APPCENTER_TRIGGER" --locale $LOCALE --app-path $APPPATH --build-dir $BUILDDIR --async --uitest-tools-dir $APPCENTER_SOURCE_DIRECTORY/packages/Xamarin.UITest.*/tools --token $APPCENTER_TOKEN
```
Comment (maxxkrakoa) [inline sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh]: Extract `ui-test-tools-dir` into a variable and similar to the others above.

File: sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+echo "Found UI tests projects:"
+find $APPCENTER_SOURCE_DIRECTORY -regex '*.UITest.*\.csproj' -exec echo {} \;
+echo
+echo "Building UI test projects:"
+find $APPCENTER_SOURCE_DIRECTORY -name '*.UITest.csproj' -exec msbuild {} \;
+echo "Compiled projects to run UI tests:"
+find $APPCENTER_SOURCE_DIRECTORY -regex '*.bin.*UITest.*\.dll' -exec echo {} \;
+#echo "Running UI test in App Center Test:"
+echo "What is in the output directory"
+ls $APPCENTER_OUTPUT_DIRECTORY
```
Comment (maxxkrakoa) [inline sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh]: Is there a specific reason for listing those files? Would it make more sense to add a check for something and exit if it doesn't work?

**Comment by King-of-Spades** [comment]
I compared your approach to my working sample here: https://github.com/King-of-Spades/AppCenter-Samples/blob/master/Xamarin.UITest/FormsGallery/FormsGallery/FormsGallery.Android/appcenter-post-build.sh

I'm not too familiar with regex stuff, so I can't evaluate that at a glance, but if the script works, then it should be fine as far as I know. 

This really needs a readme though, because some of your environment variables are custom and not defined within the script itself, so users who don't already understand how scripts work would have no way to interpret them.

File: sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+echo "Found UI tests projects:"
+find $APPCENTER_SOURCE_DIRECTORY -regex '*.UITest.*\.csproj' -exec echo {} \;
+echo
+echo "Building UI test projects:"
+find $APPCENTER_SOURCE_DIRECTORY -name '*.UITest.csproj' -exec msbuild {} \;
```
Comment (DennisPan) [inline sample-build-scripts/test/xamarin-uitest-android/appcenter-post-build.sh]: Why the regex is different for finding and building UI test projects?

**Comment by Oddj0b** [comment]
@King-of-Spades  @DennisPan @maxxkrakoa 

Thanks for the reviews, I've updated the scripts to my best understanding

**Comment by King-of-Spades** [comment]
Did a slight tweak of the 'readme' section at the top to make it more accurate & complete. Everything I can evaluate here looks good to me now (I still don't have much regex knowledge, so I haven't QA'd that section.)

**Pull request merged**